explore/pca-visualization/PCA_Visualization.py

```diff
@@ -0,0 +1,271 @@
+        ax.legend(loc='best')
+    plt.xlabel("Principal Component %s " %str(component1+1))
+    plt.ylabel("Principal Component %s " %str(component2+1))
+    plt.title("{} (PC{} and PC{})".format(title,component1+1,component2+1))
+    if len(attributes) > 12:
+        numberColumns = 2
+    else:
+        numberColumns = 1
+    plt.legend(bbox_to_anchor=(1.05, 1), loc=2, ncol=numberColumns, borderaxespad=0.)
+    plt.show()
+    
+def PlotAttributesMultiPlots(LastComponent, attribute='acronym',proportionShown = 'All',pointSize=20, X=X_pca, samples=samples, colors=None):
+    """ 
+    Create multiple plots on two principal components.
+    Takes a number (LastComponent) and plots each set of two components going
+    up starting from 0,1 the 2,3... up until (LastComponent-2),(LastComponent-1)
+    """
+    OddList = [x for x in range(LastComponent) if x % 2 == 0]
+    for i in OddList:
+        PlotAttributes(i,i+1, attribute,proportionShown,pointSize, X=X_pca, samples=samples, colors=colors)
+
+
+# In[11]:
+
+PlotAttributes(0,1,attribute='acronym')
+
+
+# In[12]:
+
+PlotAttributes(0,1,attribute='organ_of_origin')
+
+
+# ### First ten principal components relative to disease
+# ### (There looks to be a strong correlation)
+# 
+# ### TODO: Print the correlation matrix
+# 
+
+# In[13]:
+
+PlotAttributesMultiPlots(10,attribute='acronym',proportionShown=0.4)
+
+
+# ### First 10 components relative to gender
+
+# In[14]:
+
+PlotAttributesMultiPlots(10,attribute='gender',proportionShown=0.3)
+
+
+# ## 5. Visualize the relationship between gender and expression apart from disease
+# 
+# Select a few diseases that are fairly gender neutral run pca on only samples in those diseases and plot gender against a few Principal Components for each
+# 
+# (As shown by the plots below, there's still no easy to see correlation between expression principal components and gender even when disease is removed from the mix. I'm suprised that each disease seems to have a similar two clusters when the first two principal components are plotted)
+
+# In[15]:
+
+# Get a list of the gender neutral diseases (no more than 70% a particular gender)
+diseaseDict = dict.fromkeys(np.unique(samples['acronym']))
+# Create a dictionary with disease: [ration male, number of samples]
+for disease in diseaseDict:
+    tempDF = samples.loc[samples['acronym'] == disease]
+    males = tempDF['gender'].value_counts()[0]
+    genderRatio = males / tempDF.shape[0]
+    diseaseDict[disease] = [genderRatio, tempDF.shape[0]]
+# Remove the non gender neutral diseases
+neutralDiseaseDict = dict(diseaseDict)
+for disease in diseaseDict:
+    if 0.3 < diseaseDict[disease][0] < 0.7  :
+        pass
+    else:
+        del neutralDiseaseDict[disease]
+neutralDiseaseList = list(neutralDiseaseDict.keys())
+# Print the list for reference
+print(neutralDiseaseList)
```

Comment:
A webapp with visualization methods sounds intriguing!

I'll include `BRCA` when I address the other comments. Any preference on if I should just add it to the list or plot as it's own cell?

`TGCT` would be a little trickier (I assume it's only males). It might be interesting to see the PCs plotted with the total Y chromosome expression as point size or gray-scale but I think this is beyond the scope of this notebook.